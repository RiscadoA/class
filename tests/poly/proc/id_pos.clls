type Id(T) {
    recv ~T; send T; close
};;

proc id<T>(x: Id(T)) {
    println("id recv");
    recv x(y);
    println("id send");
    send x(y);
    println("id close");
    close x
};;

proc main() {
    cut {
        id<close>(id)
        |id: ~Id(close)|
        println("send");
        send id(x. println("close x"); close x);
        println("recv");
        recv id(x);
        println("wait id");
        wait id;
        println("wait x");
        wait x;
        ()
    }
};;
