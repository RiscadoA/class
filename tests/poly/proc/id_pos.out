send
id recv
close x
id send
id close
recv
wait id
wait x
